internal/application/callback.go

```diff
@@ -113,8 +113,14 @@ func UpdateDevice(updateDeviceRequest requests.UpdateDeviceRequest, dic *di.Cont
 		return errors.NewCommonEdgeX(errors.KindServerError, errMsg, err)
 	}
 
-	lc.Debugf("starting AutoEvents for device %s", device.Name)
-	container.ManagerFrom(dic.Get).RestartForDevice(device.Name)
+	autoEventManager := container.ManagerFrom(dic.Get)
+	if device.AdminState == models.Locked {
```

Comment:
Should we check if Admin state changed before doing these actions? 
i.e was already lock so autoevents is already stopped or **more importantly** was already unlocked so autoevents are already running.

Reply:
@lenny-intel 
It's fine to stop or restart no matter it was LOCKED or UNLOCKED.  The original behavior is Restart AutoEvent anyway, because the Device update should reset its AutoEvents.  It's hard to tell which update behavior impact the AutoEvents, so always restart is simple.
stop is just marking the flag of the AutoEvents.
https://github.com/edgexfoundry/device-sdk-go/blob/95a9d1abedb9d7c119ffb1f24142991202c879de/internal/autoevent/executor.go#L138-L140

Re